### ISSUE TITLE
249: Split Faces with Panel Texture

### ISSUE THREAD

**Issue by tngreene** [X-Plane/XPlane2Blender#447]
Panels are very important to some, and it would be an extremely difficult thing to redo some of them (think of old fashion dials, 100's of LEDs etc).

- [x] Split faces with panel texture into their own object, with their own special material, which has the panel information - ugg.
- [x] Code
- [x] Unit Test

**Comment by tngreene** [comment]
# Notes

## OBJ8 spec
### ATTR_cockpit

ATTR_cockpit sets the current texture to the panel texture and the current manipulator to the panel-click manipulator.

ATTR_no_cockpit - resets the back to normal.
```

It really ought to be called ATTR_panel instead.

ATTR_cockpit_region <region number>

```
[New in 900:] The ATTR_cockpit_region attribute changes the texture of the object to a specific region of the cockpit texture. Use ATTR_no_cockpit when done. The current manipulation is set to panel clicking.
```

This is written in 2.49 under the following conditions:
- ispanelok (aka iscockpit) and TEX pressed and a face's image is in exporter's regions
- ispanelokay(aka iscockpit) and TEX pressed and a face's image has 'panel.' (notice that this could match "panel.dumb.test.whatever" as well as "MyPlane_panel.png") in the name,

The regions and other information come from PanelRegionHandler, a special Object created and managed by the X-Plane panel regions

**Comment by tngreene** [comment]
We're also going to need to split meshes not only by face buttons, but also by Panel Texture or not. See #425's and #348 split goals

**Comment by tngreene** [comment]
Closed with b8201171c13!
